junit-jupiter-api/src/main/java/org/junit/jupiter/api/MethodOrderer.java

```diff
@@ -172,6 +175,74 @@ private static int getOrder(MethodDescriptor descriptor) {
 		}
 	}
 
+	/**
+	 * {@code MethodOrderer} that sorts methods based on the {@link DependsOn @DependsOn}
+	 * annotation.
+	 */
+	class DependsOnAnnotation implements MethodOrderer {
+		private static final Logger logger = LoggerFactory.getLogger(Random.class);
+
+		@Override
+		public void orderMethods(MethodOrdererContext context) {
+			// Directed Acyclic Graph to represent order relationship between methods
+			// An edge from A -> B means method A should run after B
+			Map<String, String[]> digraph = context.getMethodDescriptors().stream().filter(
+				descriptor -> descriptor.isAnnotated(DependsOn.class)).collect(
+					toMap(descriptor -> descriptor.getMethod().getName(),
+						descriptor -> descriptor.findAnnotation(DependsOn.class).map(DependsOn::value).get()));
+
+			// Give each an @Order's value equivalent to its number of previous dependencies
+			Map<String, Integer> dependencySize = new HashMap<>();
+
+			try {
+				// loop through all vertexes (methods) in graph
+				digraph.keySet().forEach(name -> {
```

Comment:
You can replace this with the following one-liner.

```java
digraph.keySet().forEach(name -> depthFirstSearch(name, digraph, dependencySize));
```

Reply:
Done.